### ISSUE TITLE
performance regression `dag(::ITensor)`

### ISSUE THREAD

**Issue by orialb** [ITensor/ITensors.jl#198]
Seems that `dag(t::ITensor)` became slower. using 
```
i = Index(1000)
t = randomITensor(ComplexF64, i)
@btime dag($t)
```
I get ` 1.272 ms (8023 allocations: 156.91 KiB)` on master, and `2.596 μs (10 allocations: 16.06 KiB)` on f92d4ab . This seems to have large effect on my TDVP code as I see that Krylov based exponentiation using `KrylovKit.exponentiate` is spending most of its time in `dag` (I'm not using this code in production yet, just wanted to point out that this slowness of `dag` can have a large effect in user codes).

I think the problem is similar to #184 where calling `conj(tensor(t))` became slower. Indeed 
On master:
```
julia> @btime conj(tensor(t));
  1.272 ms (8023 allocations: 156.91 KiB)
```
on f92d4ab:
```
julia> @btime conj(tensor(t))
  2.502 μs (7 allocations: 15.94 KiB)
```

Also here calling `conj(store(t))` (on master) is much faster compared to `conj(tensor(t)` and I get `1.747 μs (1 allocation: 15.75 KiB)` . Underlying problem might be related to `getindex` regression I pointed out in #184. 
I could make a PR with a fix for this (overloading `conj`).

**Comment by mtfishman** [comment]
Thanks for pointing this out, sorry this issue keeps popping up. Definitely need to look into why `getindex` seems to be so slow for Tensor types. Your suggestion of overloading `conj` is a good intermediate solution.

**Comment by orialb** [comment]
fixed by #201 .
